src/compiler/typescript_to_rust/azle_generate/src/candid_type/record/mod.rs

```diff
@@ -16,25 +19,41 @@ use super::errors::WrongEnclosedType;
 
 impl SourceMapped<'_, TsTypeAliasDecl> {
     pub fn to_record(&self) -> Result<Option<Record>, Vec<Error>> {
-        self.process_ts_type_ref("Record", |type_ref| {
+        self.process_ts_type_ref(&self.symbol_table.record, |type_ref| {
             let (type_params, members) =
                 (self.get_type_params(), type_ref.to_record()).collect_results()?;
-            Ok(Record {
-                name: Some(self.id.get_name().to_string()),
-                type_params: type_params.into(),
-                ..members
-            })
+            match members {
+                Some(members) => Ok(Some(Record {
+                    name: Some(self.id.get_name().to_string()),
+                    type_params: type_params.into(),
+                    ..members
+                })),
+                None => Ok(None),
+            }
         })
+        .map(|result| result.flatten())
     }
 }
 
 impl SourceMapped<'_, TsTypeRef> {
-    pub fn to_record(&self) -> Result<Record, Vec<Error>> {
-        match self.get_ts_type()?.as_ts_type_lit() {
-            Some(ts_type_lit) => ts_type_lit,
-            None => return Err(vec![WrongEnclosedType::error_from_ts_type_ref(self).into()]),
+    pub fn to_record(&self) -> Result<Option<Record>, Vec<Error>> {
```

Comment:
Revert

Reply:
We decided to leave Result<Option> with the thought that we might change to Option<Result>